src/compile/scale/range.ts

```diff
@@ -229,6 +256,39 @@ export function defaultRange(
   throw new Error(`Scale range undefined for channel ${channel}`);
 }
 
+function defaultContinuousToDiscreteCount(scaleType: ScaleType, config: Config, domain: Domain) {
+  switch (scaleType) {
+    case 'quantile':
+      return config.scale.quantileCount;
+    case 'quantize':
+      return config.scale.quantizeCount;
+    case 'threshold':
+      if (domain !== undefined && isArray(domain)) {
+        return domain.length + 1;
+      } else {
+        return 4;
```

Comment:
I wonder if returning 4 makes sense at all.  
(Shouldn't we throw warning that an array domain must be specified?)

Reply:
We can also throw a warning and still returns 4?

Reply:
I think you have to test a bit.  What's the default number of threshold level there is no specified domain?  The default number here should match it.

Reply:
As far as I know, there's no default range threshold in vega. If no domain and range is specified for threshold scale, vega will display one empty legend with `>=0.5` label